src/client/components/BookingForms/AddMeal.js

```diff
@@ -0,0 +1,149 @@
+import React, { useState } from "react";
+import { useHistory } from "react-router-dom";
+import "./forms.css";
+import { useForm } from "react-hook-form";
+import { joiResolver } from "@hookform/resolvers/joi";
+import Joi from "joi";
+
+const mealsValidSchema = Joi.object({
+  id: Joi.number().integer(),
+  title: Joi.string().required(),
+  description: Joi.string(),
+  location: Joi.string().required(),
+  created_date: Joi.date(),
+  Hosting_time: Joi.date().greater("now").required(),
+  max_number_of_guests: Joi.number().integer().min(10).max(100).required(),
+  price: Joi.number().precision(2).required(),
+});
+
+function AddMeal({ closePopUp }) {
+  const [title, setTitle] = useState("");
+  const [servingNum, setServingNum] = useState("");
+  const [price, setPrice] = useState("");
+  const [description, setDescription] = useState("");
+  const [location, setLocation] = useState("");
+  const [eventDate, setEventDate] = useState("");
+  const history = useHistory();
+
+  const { register, handleSubmit } = useForm({
+    resolver: joiResolver(mealsValidSchema),
+  });
+
+  const submitMeal = (e) => {
+    const meal = {
+      title,
+      description,
+      location,
+      Hosting_time: eventDate,
+      max_number_of_guests: servingNum,
+      price,
+    };
+    console.log(meal);
+    fetch("/api/meals", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(meal),
+    }).then((response) => {
+      if (response.status >= 200 && response.status <= 299) {
+        return alert(`${meal} successfully added. thanks for sharing.`);
+      } else {
+        alert(
+          "Error occurred , please make sure , that the input data type are correct "
+        );
+        throw Error(response.statusText);
```

Comment:
I'd expect the page to show an error message within the page instead of alert + throw.